module/zfs/zfs_vnops.c

```diff
@@ -326,7 +326,7 @@ zfs_write(znode_t *zp, uio_t *uio, int ioflag, cred_t *cr)
 	if (n == 0)
 		return (0);
 
-	rlim64_t limit = MAXOFFSET_T;
+	const rlim64_t limit = MAXOFFSET_T;
```

Comment:
Do we need this variable at all?

Reply:
No. It should just be GC'd.

Reply:
It's used further down in the function, so it could be moved closer for clarity:
```C
        const rlim64_t limit = MAXOFFSET_T;

        if (woff >= limit) {
                zfs_rangelock_exit(lr);
                ZFS_EXIT(zfsvfs);
                return (SET_ERROR(EFBIG));
        }

        if ((woff + n) > limit || woff > (limit - n))
                n = limit - woff;
```

Reply:
```C
        if ((woff + n) > limit || woff > (limit - n))
                n = limit - woff;
```
That looks like it could be simplified a bit, too. I can take care of these in a followup.

Reply:
34e7989